apps/rush/src/actions/CustomRushAction.ts

```diff
@@ -0,0 +1,174 @@
+      () => {
+        stopwatch.stop();
+        console.log(colors.green(`rush ${this.options.actionVerb} (${stopwatch.toString()})`));
+        this._collectTelemetry(stopwatch, true);
+        this._parser.flushTelemetry();
+        this.eventHooksManager.handle(Event.postRushBuild, this._parser.isDebug);
+      },
+      () => {
+        stopwatch.stop();
+        console.log(colors.red(`rush ${this.options.actionVerb} - Errors! (${stopwatch.toString()})`));
+        this._collectTelemetry(stopwatch, false);
+        this._parser.flushTelemetry();
+        this.eventHooksManager.handle(Event.postRushBuild, this._parser.isDebug);
+        this._parser.exitWithError();
+      });
+  }
+
+  protected onDefineParameters(): void {
+    this._parallelismParameter = this.defineIntegerParameter({
+      parameterLongName: '--parallelism',
+      parameterShortName: '-p',
+      key: 'COUNT',
+      description: 'Change limit the number of simultaneous builds. This value defaults to the number of CPU cores'
+    });
+    this._toFlag = this.defineStringListParameter({
+      parameterLongName: '--to',
+      parameterShortName: '-t',
+      key: 'PROJECT1',
+      description: 'Build the specified project and all of its dependencies'
+    });
+    this._fromFlag = this.defineStringListParameter({
+      parameterLongName: '--from',
+      parameterShortName: '-f',
+      key: 'PROJECT2',
+      description: 'Build all projects that directly or indirectly depend on the specified project'
+    });
+    this._verboseParameter = this.defineFlagParameter({
+      parameterLongName: '--verbose',
+      parameterShortName: '-v',
+      description: 'Display the logs during the build, rather than just displaying the build status summary'
+    });
+
+    // @TODO we should throw if they are trying to overwrite built in flags
+
+    this.customOptions.forEach((customOption: ICustomOptionInstance) => {
+      if (customOption.optionType === 'flag') {
+        customOption.parameterValue = this.defineFlagParameter({
+          parameterShortName: customOption.shortName,
+          parameterLongName: customOption.longName,
+          description: customOption.description
+        });
+      } else if (customOption.optionType === 'enum') {
+        customOption.parameterValue = this.defineOptionParameter({
+          parameterShortName: customOption.shortName,
+          parameterLongName: customOption.longName,
+          description: customOption.description,
+          defaultValue: (customOption as ICustomOption as ICustomEnumOption).defaultValue,
+          options: (customOption as ICustomOption as ICustomEnumOption).enumValues.map(
+            (enumValue: ICustomEnumValue) => {
+              return enumValue.name;
+            })
+        });
+      }
+    });
+  }
+
+  private _collectTelemetry(stopwatch: Stopwatch, success: boolean): void {
+    const extraData: { [key: string]: string } = {
+      to: (!!this._toFlag.value).toString(),
```

Comment:
I am thinking of prefixing the extraData fields to become "build_to", "build_from", and the new" build + customOption.longName". Can you include that change in this pullrequest?